pennylane/tape/tape.py

```diff
@@ -163,7 +163,7 @@ def expand_tape(tape, depth=1, stop_at=None, expand_measurements=False):
     if tape._obs_sharing_wires:
         try:
             rotations, diag_obs = qml.grouping.diagonalize_qwc_pauli_words(tape._obs_sharing_wires)
-        except ValueError as e:
+        except (TypeError, ValueError) as e:
```

Comment:
When we have `return expval(H1), expval(H2)` then `_obs_sharing_wires` are Hamiltonians, in which case a `TypeError` is returned because Pauli grouping cannot handle them.